sdk/include/psdk/fltdefs.h

```diff
@@ -0,0 +1,270 @@
+#define FD_FLAGS_ALLFLAGS     FD_FLAGS_NOSYN
+
+#define FILTER_PROTO(ProtoId)  MAKELONG(MAKEWORD((ProtoId), 0x00), 0x00000)
+#define FILTER_PROTO_ANY       FILTER_PROTO(0x00)
+#define FILTER_PROTO_ICMP      FILTER_PROTO(0x01)
+#define FILTER_PROTO_TCP       FILTER_PROTO(0x06)
+#define FILTER_PROTO_UDP       FILTER_PROTO(0x11)
+
+#define FILTER_TCPUDP_PORT_ANY ((WORD)0x0000)
+#define FILTER_ICMP_TYPE_ANY   ((BYTE)0xff)
+#define FILTER_ICMP_CODE_ANY   ((BYTE)0xff)
+
+#define LB_SRC_ADDR_USE_SRCADDR_FLAG 0x00000001
+#define LB_SRC_ADDR_USE_DSTADDR_FLAG 0x00000002
+#define LB_DST_ADDR_USE_SRCADDR_FLAG 0x00000004
+#define LB_DST_ADDR_USE_DSTADDR_FLAG 0x00000008
+#define LB_SRC_MASK_LATE_FLAG        0x00000010
+#define LB_DST_MASK_LATE_FLAG        0x00000020
+
+typedef PVOID FILTER_HANDLE, * PFILTER_HANDLE;
+typedef PVOID INTERFACE_HANDLE, * PINTERFACE_HANDLE;
+
+typedef enum _GlobalFilter
+{
+    GF_FRAGMENTS = 2,
+    GF_STRONGHOST = 8,
+    GF_FRAGCACHE = 9
+} GLOBAL_FILTER, *PGLOBAL_FILTER;
+
+typedef enum _PfAddresType
+{
+    PF_IPV4,
+    PF_IPV6
+} PFADDRESSTYPE, *PPFADDRESSTYPE;
+
+typedef enum _PfForwardAction
+{
+    PF_ACTION_FORWARD = 0,
+    PF_ACTION_DROP
+} PFFORWARD_ACTION, *PPFFORWARD_ACTION;
+
+typedef enum _PfFrameType
+{
+    PFFT_FILTER = 1,
+    PFFT_FRAG = 2,
+    PFFT_SPOOF = 3
+} PFFRAMETYPE, *PPFFRAMETYPE;
+
+typedef struct _pfLogFrame
+{
+    LARGE_INTEGER Timestamp;
+    PFFRAMETYPE pfeTypeOfFrame;
+    DWORD dwTotalSizeUsed;
+    DWORD dwFilterRule;
+    WORD wSizeOfAdditionalData;
+    WORD wSizeOfIpHeader;
+    DWORD dwInterfaceName;
+    DWORD dwIPIndex;
+    BYTE bPacketData[1];
+} PFLOGFRAME, *PPFLOGFRAME;
+
+typedef struct _PF_FILTER_DESCRIPTOR
+{
+    DWORD dwFilterFlags;
+    DWORD dwRule;
+    PFADDRESSTYPE pfatType;
+    PBYTE SrcAddr;
+    PBYTE SrcMask;
+    PBYTE DstAddr;
+    PBYTE DstMask;
+    DWORD dwProtocol;
+    DWORD fLateBound;
+    WORD wSrcPort;
+    WORD wDstPort;
+    WORD wSrcPortHighRange;
+    WORD wDstPortHighRange;
+} PF_FILTER_DESCRIPTOR, *PPF_FILTER_DESCRIPTOR;
+
+#define FILTERSIZE (sizeof(PF_FILTER_DESCRIPTOR) - (DWORD)FIELD_OFFSET(PF_FILTER_DESCRIPTOR, SrcAddr))
+
+typedef struct _PF_FILTER_STATS
+{
+    DWORD dwNumPacketsFiltered;
+    PF_FILTER_DESCRIPTOR info;
+} PF_FILTER_STATS, *PPF_FILTER_STATS;
+
+typedef struct _PF_INTERFACE_STATS
+{
+    PVOID pvDriverContext;
+    DWORD dwFlags;
+    DWORD dwInDrops;
+    DWORD dwOutDrops;
+    PFFORWARD_ACTION eaInAction;
+    PFFORWARD_ACTION eaOutAction;
+    DWORD dwNumInFilters;
+    DWORD dwNumOutFilters;
+    DWORD dwFrag;
+    DWORD dwSpoof;
+    DWORD dwReserved1;
+    DWORD dwReserved2;
+    LARGE_INTEGER liSYN;
+    LARGE_INTEGER liTotalLogged;
+    DWORD dwLostLogEntries;
+    PF_FILTER_STATS FilterInfo[1];
+} PF_INTERFACE_STATS, *PPF_INTERFACE_STATS;
+
+typedef struct _PF_LATEBIND_INFO
+{
+    PBYTE SrcAddr;
+    PBYTE DstAddr;
+    PBYTE Mask;
+} PF_LATEBIND_INFO, *PPF_LATEBIND_INFO;
+
+DWORD
+PFEXPORT
+WINAPI
```

Comment:
Why are these first two not `PFAPIENTRY`?

Reply:
I first thought about using this to be more explicit (I don't really like the macro containing everything), but I discarded the idea, yet forgot to revert these ones back.